src/entity/Entity.cpp

```diff
@@ -215,26 +215,55 @@ bool Entity::init(AttributeMap &overrides,
     ////////////////////////////////////////////////////////////
     // controller
     ////////////////////////////////////////////////////////////
-    auto it = info.find("controller");
-    if (it == info.end()) {
-        controller_ = std::make_shared<Controller>();
-    } else {
-        controller_ = init_controller(it->second, overrides["controller"], motion_model_->vars());
-    }
+    int controller_ct = 0;
+    std::string controller_name = std::string("controller") + std::to_string(controller_ct);
+
+    while (info.count(controller_name) > 0) {
+        ControllerPtr controller = init_controller(info[controller_name],
+                                                   overrides[controller_name],
+                                                   motion_model_->vars());
+        if (controller == nullptr) {
+            std::cout << "Failed to open controller plugin: " << controller_name << std::endl;
+            return false;
+        }
 
-    if (controller_ == nullptr) {
-        std::cout << "Failed to open controller plugin: " << it->second << std::endl;
-        return false;
+        if (!verify_io_connection(controller->vars(), motion_model_->vars())) {
```

Comment:
Having this check for every controller's connection to the motion model seems like it may raise some false positives. For instance, if we want the serial connection controller1 -> controller2 -> motion, then this is going to check that controller1 ->motion is connected correctly.

Reply:
Yes, I agree. I moved that verify_io_connection to where the last controller is connected to the motion model.